src/Components/FollowButton/FollowGeneButton.tsx

```diff
@@ -112,13 +114,49 @@ export const FollowGeneButtonFragmentContainer = createFragmentContainer(
   FollowGeneButton,
   {
     gene: graphql`
-      fragment FollowGeneButton_gene on Gene {
+      fragment FollowGeneButton_gene on Gene
+        @argumentDefinitions(
+          isLoggedIn: { type: "Boolean", defaultValue: false }
+        ) {
         id
         slug
         name
         internalID
-        isFollowed
+        isFollowed @include(if: $isLoggedIn)
```

Comment:
In general this is kind of a nice trick. The issue was 'oh maybe we can skip the follow button query completely when logged out'.

But...if you just had a query renderer with an early return when logged out, you don't get the right follow button rendering (right?). As in, when you click the button logged out it should pop up the auth dialog, etc. - it's not _just_ a completely vanilla button.

So, you can _maybe_ drop this request, but instead of an early return, you'd still need to return a proper button component that pops up the auth dialog, also some analytics tracking on clicks? And so forth.

Whereas here, this is an early return in MP as well - https://github.com/artsy/metaphysics/blob/9ff29dc74c5e5d1937bf9ab089939536259296a8/src/schema/v2/gene.ts#L141 - so skipping it here isn't really _that_ critical.

Really - I guess what I'm thinking/trying to say - is the original ticket said to suppress this call altogether when the user is logged out, and that's what I had in my head - that this query wouldn't happen at all when logged out.

But actually - this setup reaffirms that we should consider the follow button as a little component that _does_ require some data, even to setup when logged out (like name, id/slug) - and so, it _should_ define its own fragment and fire a query no matter what.

And at that point, we might as well use `@include` here for this field (vs. rely on MP early return).

Sorry for the long-winded comment! Just mentally probably going through the thinking steps you've gone through here.

Does this seem correct to you? **All follow button components should be in their own query renderers, always firing whether you are logged in/out, _and_ for extra credit you can use `@include` on the `isFollowed` field**

Reply:
Thanks for this @mzikherman. It is nice to understand the reasoning 🚀 
Definitely makes sense, I'll proceed with other follow buttons in the upcoming PRs!